staging/src/k8s.io/apiserver/pkg/authorization/authorizer/interfaces.go

```diff
@@ -80,7 +80,7 @@ func (f AuthorizerFunc) Authorize(ctx context.Context, a Attributes) (Decision,
 // RuleResolver provides a mechanism for resolving the list of rules that apply to a given user within a namespace.
 type RuleResolver interface {
 	// RulesFor get the list of cluster wide rules, the list of rules in the specific namespace, incomplete status and errors.
-	RulesFor(user user.Info, namespace string) ([]ResourceRuleInfo, []NonResourceRuleInfo, bool, error)
+	RulesFor(ctx context.Context, user user.Info, namespace string) ([]ResourceRuleInfo, []NonResourceRuleInfo, bool, error)
```

Comment:
since context can inherently contain a namespace, it is confusing what this context will be used for... timeout/tracing only, or more than that? the explicit namespace passed in should be used for the lookup, not any coordinate contained in the context.

docs need to be super explicit about what the context will be used for so we don't get mismatches or something downstream pulling coordinate information out of the context

Reply:
a similar comment applies to every function that this PR is adding a context to which also takes parameters (namespace, authz attributes, etc) which can overlap with or conflict with information in the context

Reply:
As these functions historically didn't have context wired in, context didn't exist there. 
Would document updating for each function name would be sufficient? 

Or to be super sure we could add a helper:

`ctx := genericapirequest.WithoutNamespace(ctx)` which would remove the namespace if it's present from functions where we wire in the context now.  This way tracing, timeouts, still works and we safeguard this from in-intentional behaviours.

Reply:
I think documentation is necessary, but not sufficient. When we plumb context through to lower layers, we also need to ensure the coordinate objects aren't plumbed down.

The two coordinate objects I know of embedded into context are the namespace and requestInfo (which duplicates the namespace). WithoutNamespace/WithoutRequestInfo at the point where we propagate this to something like etcd makes this easier to reason about

Reply:
Make sense. Gonna see how it looks like with added helpers.